cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceContext.java

```diff
@@ -82,7 +85,7 @@ public BasicMapReduceContext(Program program,
                                MetricsCollectionService metricsCollectionService,
                                DatasetFramework dsFramework,
                                CConfiguration conf) {
-    super(program, runId, runtimeArguments, datasets,
+    super(program, runId, extractScope(program.getName(), runtimeArguments), datasets,
```

Comment:
Just figured out that the BasicArguments ctor makes an immutable copy. So please disregard this